sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTestBase.java

```diff
@@ -0,0 +1,605 @@
+                } else {
+                    result = asyncClient.analyze(
+                        BinaryData.fromFile(new File(imageSource).toPath()),
+                        visualFeatures,
+                        options).block();
+                }
+            } catch (HttpResponseException e) {
+                System.out.println("Expected exception: " + e.getMessage());
+                assertEquals(e.getResponse().getStatusCode(), expectedStatusCode);
+                assertTrue(e.getMessage().contains(expectedMessageContains));
+                return;
+            }
+        }
+
+        System.out.println("Test should have thrown an exception, but it did not");
+        assertTrue(false);
+    }
+
+    private static void validateAnalysisResult(
+        ImageAnalysisResult result,
+        List<VisualFeatures> expectedFeatures,
+        Boolean genderNeutralCaption,
+        List<Double> aspectRatios) {
+
+        validateMetadata(result);
+        validateModelVersion(result);
+
+        if (expectedFeatures != null && expectedFeatures.contains(VisualFeatures.CAPTION)) {
+            validateCaption(result, genderNeutralCaption);
+        } else {
+            assertNull(result.getCaption());
+        }
+
+        if (expectedFeatures != null && expectedFeatures.contains(VisualFeatures.DENSE_CAPTIONS)) {
+            validateDenseCaptions(result);
+        } else {
+            assertNull(result.getDenseCaptions());
+        }
+
+        if (expectedFeatures != null && expectedFeatures.contains(VisualFeatures.OBJECTS)) {
+            validateObjects(result);
+        } else {
+            assertNull(result.getObjects());
+        }
+
+        if (expectedFeatures != null && expectedFeatures.contains(VisualFeatures.TAGS)) {
+            validateTags(result);
+        } else {
+            assertNull(result.getTags());
+        }
+
+        if (expectedFeatures != null && expectedFeatures.contains(VisualFeatures.PEOPLE)) {
+            validatePeople(result);
+        } else {
+            assertNull(result.getPeople());
+        }
+
+        if (expectedFeatures != null && expectedFeatures.contains(VisualFeatures.SMART_CROPS)) {
+            validateSmartCrops(result, aspectRatios);
+        } else {
+            assertNull(result.getSmartCrops());
+        }
+
+        if (expectedFeatures != null && expectedFeatures.contains(VisualFeatures.READ)) {
+            validateRead(result);
+        } else {
+            assertNull(result.getRead());
+        }
+    }
+
+    private static void validateMetadata(ImageAnalysisResult result) {
+        assertNotNull(result.getMetadata());
+        assertEquals(result.getMetadata().getHeight(), 576);
+        assertEquals(result.getMetadata().getWidth(), 864);
+    }
+
+    private static void validateModelVersion(ImageAnalysisResult result) {
+        assertNotNull(result.getModelVersion());
+        assertFalse(result.getModelVersion().isEmpty());
```

Comment:
I think this should be removed, if it ends up being empty this error message produced by `assertFalse` will be more confusing than `assertEquals` telling you `excpected: 2023-10-01, got: ""`

Reply:
Thanks @alzimmermsft. Agree! the line `assertFalse(result.getModelVersion().isEmpty());` will be removed in the next commit.